compiler/rustc_errors/src/lib.rs

```diff
@@ -1248,9 +1248,13 @@ impl HandlerInner {
     }
 
     fn treat_err_as_bug(&self) -> bool {
-        self.flags
-            .treat_err_as_bug
-            .map_or(false, |c| self.err_count() + self.lint_err_count >= c.get())
+        self.flags.treat_err_as_bug.map_or(false, |c| {
```

Comment:
Well, I didn't mean to include this, but this is a fix to `-Ztreat-err-as-bug` which doesn't count delayed span bugs correctly. I can remove it and put it into another PR if needed.

--

detailed explanation: If we emit an error A, delay a bug B, then emit error C. Then:
1. running `-Ztreat-err-as-bug=1` will make it ICE on A.
2. running `-Ztreat-err-as-bug=2` will make it ICE on B.
3. running `-Ztreat-err-as-bug=3` will skip over B and will make rust _skip_ over C.

So there's literally no value `N` such that `-Ztreat-err-as-bug=N` will properly work to catch C. We add the delayed good path and span bugs into the count here so this works properly now.